src/Components/CreateImageWizard/utilities/useValidation.tsx

```diff
@@ -133,6 +137,20 @@ export function useFirstBootValidation(): StepValidation {
   };
 }
 
+export function useHostnameValidation(): StepValidation {
+  const hostname = useAppSelector(selectHostname);
+
+  if (!isHostnameValid(hostname)) {
+    return {
+      errors: {
+        hostname: 'Invalid hostname',
```

Comment:
Let's provide some additional guidance to users about what constitutes a valid hostname.

Had to look this up myself, of course. Run `man 5 hostname` and you get this:
```
HOSTNAME(5)                                                                         hostname                                                                         HOSTNAME(5)

NAME
       hostname - Local hostname configuration file

SYNOPSIS
       /etc/hostname

DESCRIPTION
       The /etc/hostname file configures the name of the local system. Unless overridden as described in the next section, systemd(1) will set this hostname during boot using
       the sethostname(2) system call.

       The file should contain a single newline-terminated hostname string. Comments (lines starting with a "#") are ignored. The hostname should be composed of up to 64 7-bit
       ASCII lower-case alphanumeric characters or hyphens forming a valid DNS domain name. It is recommended that this name contains only a single label, i.e. without any
       dots. Invalid characters will be filtered out in an attempt to make the name valid, but obviously it is recommended to use a valid name and not rely on this filtering.

       You may use hostnamectl(1) to change the value of this file during runtime from the command line. Use systemd-firstboot(1) to initialize it on mounted (but not booted)
       system images.
```

So I would suggest the error say: "Invalid hostname. The hostname should be composed of up to 64 7-bit
       ASCII lower-case alphanumeric characters or hyphens forming a valid DNS domain name. It is recommended that this name contains only a single label, i.e. without any
       dots. Invalid characters will be filtered out in an attempt to make the name valid, but obviously it is recommended to use a valid name and not rely on this filtering."